cmd/s3NameResolver.go

```diff
@@ -61,49 +61,43 @@ const failToResolveMapValue = "<resolving_failed>"
 var s3BucketNameResolveError = "fail to resolve s3 bucket name"
 
 // NewS3BucketNameToAzureResourcesResolver creates S3BucketNameToAzureResourcesResolver.
-// S3BucketNameToAzureResourcesResolver works in such pattern:
-// 1. User provided all the bucket names returned in a certain time. (S3 has service limitation, that one S3 account can only have 100 buckets, except opening service ticket to increase the number.)
-// 2. S3BucketNameToAzureResourcesResolver resolves the names with one logic pass during creating the resolver instance.
-// 3. User can get resolved name later with ResolveName.
-// As S3BucketNameToAzureResourcesResolver need to detect naming collision, the resolver doesn't accept adding new bucket name except the initial s3BucketNames,
-// considering there could be future valid name that is not predictable during previous naming resolving.
+// Users can provide bucket names upfront and on-demand via ResolveName.
+// Previously resolved names will be returned outright by ResolveName.
 func NewS3BucketNameToAzureResourcesResolver(s3BucketNames []string) *S3BucketNameToAzureResourcesResolver {
 	s3Resolver := S3BucketNameToAzureResourcesResolver{
```

Comment:
Reading this prompted some more questions in my mind, about the safety of appending integers to ensure uniqueness.  Have sent you a long email with the details.  Pls leave this comment here open until the email thread is resolved.

Reply:
Thread resolved: We decided not to change the functionality at this moment.